Dungeons/Dungeons/TileContainers/DungeonNodeConnector.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Dungeons.Core;
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Dungeons.TileContainers
+{
+  public class DungeonNodeConnector
+  {
+    public Point Node1Position;
+    public Point Node2Position;
+    public Point CorridorPosition;
+    public RoomPlacement placement = new RoomPlacement();
+    public const int centralRoomPosition = 12;
+    int[] wallLength = { 0, 1, 2, 3 };
+    public int upperLenght = RandHelper.Random.Next(24, 27);
+    public int bottomLenght = RandHelper.Random.Next(24, 27);
+    public int rightLenght = RandHelper.Random.Next(24, 27);
+    public int leftLenght = RandHelper.Random.Next(24, 27);
+
+    public DungeonNode ConnectNodes(List<DungeonNode> nodes, DungeonNode node1, DungeonNode node2, DungeonGenerator dungeon)
+    {
+      Node1Position = PlaceNodes(node1);
+      Node2Position = PlaceNodes(node2);
+
+      var infoC = new GenerationInfo();
+      infoC.NumberOfRooms = 1;
+      if (node1.Placement == RoomPlacement.LeftUpper)
+      {
+        infoC.MinNodeSize = new Size(Node2Position.X - node1.Width + wallLength[3], 4);
+        infoC.MaxNodeSize = new Size(Node2Position.X - node1.Width + wallLength[3], 4);
+        CorridorPosition = new Point(node1.Width - wallLength[1], Node1Position.Y + (node1.Height / 2) - wallLength[1]);
+        placement = RoomPlacement.CorrindorHorizontalTop;
+      }
+      if (node1.Placement == RoomPlacement.RightUpper)
+      {
+        infoC.MinNodeSize = new Size(4, Node2Position.Y - node1.Height + wallLength[3]);
+        infoC.MaxNodeSize = new Size(4, Node2Position.Y - node1.Height + wallLength[3]);
+        CorridorPosition = new Point(Node1Position.X + (node1.Width / 2) - wallLength[1], node1.Height - wallLength[1]);
+        placement = RoomPlacement.CorrindorVerticalRight;
+      }
+      if (node1.Placement == RoomPlacement.RightLower)
```

Comment:
Jak by się dało to to napisać w jednym bloku z RoomPlacement.LeftUpper.
if (node1.Placement == RoomPlacement.LeftUpper || node1.Placement == RoomPlacement.RightLower)
{
  //kod wspolny dla obu
 //...
 //no i kod specyficzny dla danego przypadku: 
 if(node1.Placement == RoomPlacement.LeftUpper )
  {
  }
  else
  {
  }
}